spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/mapping/FirestorePersistentEntityImpl.java

```diff
@@ -57,4 +55,17 @@ public FirestorePersistentProperty getIdPropertyOrFail() {
 		}
 		return getIdProperty();
 	}
+
+	private static <T> String getEntityCollectionName(TypeInformation<T> typeInformation) {
+		Entity entity = AnnotationUtils.findAnnotation(typeInformation.getType(), Entity.class);
+		String collectionName = (String) AnnotationUtils.getValue(entity, "collectionName");
+
+		if (collectionName == null || collectionName.isEmpty()) {
+			// Infer the collection name as the lowercase entity name.
+			return StringUtils.uncapitalize(typeInformation.getType().getSimpleName());
```

Comment:
Whats wrong with `String.toLowerCase()`?

Reply:
This is better because if the user has an entity named `EmployeeRecord` it will only lowercase the first letter so it retains the camelcasing `employeeRecord`.

This is also the convention used in Spanner for inferring table name if the user does not explicitly set it for their entity.

Reply:
Oh!